service/src/test/java/bio/terra/javatemplate/iam/SamServiceTest.java

```diff
@@ -4,20 +4,21 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.when;
 
+import bio.terra.javatemplate.BaseSpringBootTest;
 import org.broadinstitute.dsde.workbench.client.sam.ApiException;
 import org.broadinstitute.dsde.workbench.client.sam.api.StatusApi;
 import org.broadinstitute.dsde.workbench.client.sam.model.SystemStatus;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
 
 @ExtendWith(MockitoExtension.class)
-class SamServiceTest {
+class SamServiceTest extends BaseSpringBootTest {
```

Comment:
I'm away on vacation until 10/31, but I think this is a bad idea. In general, I don't think we should use inheritance for tests. Inheritance in tests IMO makes tests harder to read and understand, and doesn't provide enough benefit to make it worthwhile. It ends up being used for code sharing, and inheritance is not a good way share code.

My thinking about spring tests is you have three kinds of tests:
1. webmvc tests
2. "pure" unit tests, where all components are mocked except for the class being tested
3. DAO tests, which require a database connection

Most tests are type (2) and shouldn't require any annotations besides `@ExtendWith(MockitoExtension.class)`. WebMvc tests only need `WebMvcTest` and `ContextConfiguration`. Dao tests only need `@SpringBootTest` and `@Transactional`.